### PR TITLE
Revive PR: Add RequestTimeout as option for RavenDB healthcheck

### LINKED ISSUE #1558: Make RavenDB Healthcheck Timeout configurable
**What would you like to be added**:
It would be great if we could have the timeout that is used for the RavenDB healthcheck configurable, right now it is hardcoded at 15 seconds.

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/4bd0e1dca241ccebb43993d21162cc02b4de7913/src/HealthChecks.RavenDB/RavenDBHealthCheck.cs#L136

**Why is this needed**:
It would be good to be able to control this timeout based on the general health check settings one wants to use.

### PR DESCRIPTION
**Pull request by cieciurm**
**What this PR does / why we need it**:
This PR revives #1707

Description
> Possibility to set a timeout for RavenDB requests
> Make RavenDB request timeout configurable for Healthcheck

**Which issue(s) this PR fixes**:
Orignal PR fixes #1558

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [ ] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by cieciurm** [comment]
I have an issue with making a test that would verify that HealthCheck failed. 

I set the timeout to be 0.001 seconds, but perhaps this is to small to be respected by the RavenDb client.

I get the following response from HC:

```json
{"status":"Healthy","totalDuration":"00:00:00.0200207","entries":{"ravendb":{"data":{},"duration":"00:00:00.0199749","status":"Healthy","tags":["ravendb"]}}}
```

So it succeeds after 0.02 seconds.

**Comment by sungam3r** [comment]
See `DatabaseHealthCheckCommand.ctor`.

**Comment by cieciurm** [comment]
After running RavenDb locally and debugging the health check - finally I have managed to adjust it correctly. Ready for review - @sungam3r 😄

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 1]
```diff
@@ -73,6 +78,7 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             });
 
             var store = value.Store;
+            SetRequestTimeoutForStoreForSpecificDatabase(store);
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:81]:
```suggestion
```

Seems as duplicate.

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 2]
```diff
@@ -106,14 +112,22 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         }
     }
 
+    private void SetRequestTimeoutForStoreForSpecificDatabase(IDocumentStore? store)
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:115]:
```suggestion
    private void SetRequestTimeoutForStoreForSpecificDatabase(IDocumentStore store)
```

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 3]
```diff
@@ -106,14 +112,22 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         }
     }
 
+    private void SetRequestTimeoutForStoreForSpecificDatabase(IDocumentStore? store)
+    {
+        if (!string.IsNullOrWhiteSpace(_options.Database))
+        {
+            store?.SetRequestTimeout(_requestTimeout, _options.Database);
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:119]:
```suggestion
            store.SetRequestTimeout(_requestTimeout, _options.Database);
```

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 4]
```diff
@@ -123,21 +137,28 @@ private static async Task CheckDatabaseHealthAsync(IDocumentStore store, string
             return;
         }
 
-        await executor.SendAsync(_databaseHealthCheck, cancellationToken).ConfigureAwait(false);
+        await executor.SendAsync(new DatabaseHealthCheckOperation(_requestTimeout), cancellationToken).ConfigureAwait(false);
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:140]:
```suggestion
        await executor.SendAsync(new DatabaseHealthCheckOperation(_options.RequestTimeout ?? TimeSpan.FromSeconds(DEFAULT_REQUEST_TIMEOUT_IN_SECONDS)), cancellationToken).ConfigureAwait(false);
```

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 5]
```diff
@@ -51,6 +53,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                 try
                 {
                     store.Initialize();
+                    _requestTimeout = _options.RequestTimeout ?? TimeSpan.FromSeconds(DEFAULT_REQUEST_TIMEOUT_IN_SECONDS);
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:56]:
```suggestion
```

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 6]
```diff
@@ -17,16 +17,18 @@ namespace HealthChecks.RavenDB;
...
     private static readonly GetBuildNumberOperation _serverHealthCheck = new();
 
-    private static readonly DatabaseHealthCheckOperation _databaseHealthCheck = new();
-
     private static readonly GetStatisticsOperation _legacyDatabaseHealthCheck = new();
 
     private static readonly ConcurrentDictionary<RavenDBOptions, DocumentStoreHolder> _stores = new();
 
+    private TimeSpan _requestTimeout;
+
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:31]:
```suggestion
```

File: src/HealthChecks.RavenDB/RavenDBHealthCheck.cs [hunk 7]
```diff
@@ -106,14 +108,22 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         }
     }
 
+    private void SetRequestTimeoutForStoreForSpecificDatabase(IDocumentStore store)
```
Comment (sungam3r) [inline src/HealthChecks.RavenDB/RavenDBHealthCheck.cs:111]: I would inline that method.

**Review by sungam3r — APPROVED** [review]

**Pull request merged**